pkg/plugins/gateway/gateway.go

```diff
@@ -156,6 +167,48 @@ func (s *Server) selectTargetPod(ctx *types.RoutingContext, pods types.PodList)
 	return router.Route(ctx, &utils.PodArray{Pods: readyPods})
 }
 
+// validateHTTPRouteStatus checks if httproute object exists and validates its conditions are true
+func (s *Server) validateHTTPRouteStatus(ctx context.Context, model string) string {
```

Comment:
why does this method return a string? does error look better? the caller can use  error.Error() to wrap the message, otherwise, for validated scenarios, it need to check string == ""? 

I feel change to error type as return result is better

Reply:
updated validate method to return error.